src/core/state/backendNetworks/backendNetworks.ts

```diff
@@ -160,6 +235,12 @@ export const useBackendNetworksStore = createQueryStore<
     setData: ({ data, set }) => {
       set((state) => {
         if (isEqual(state.backendNetworks, data)) return state;
+
+        void Promise.all([
+          handleUpdateUserChains(data.networks),
+          handleDiffAndMergeRainbowChains(data.networks),
+        ]);
+
         return {
           backendChains: transformBackendNetworksToChains(data.networks),
```

Comment:
Could be a bottleneck, but I like the idea of only storing `backendNetworks` so that we can do transform at boot-time. Wagmi/Viem have had a few breaking changes with their `Chain` type that I can remember, so we'd need to catch those type differences before hydrating `backendChains`, and then do a migration to re-transform in the state store. Whereas "dumb" backend caching here inherently requires type validity upon hydration